### PR TITLE
issue #61 changes to allow user to set the name of the output file when

### PR DESCRIPTION
**Pull request by rewopkram**
invoking Explore graph paths menu option

### CODE REVIEW

File: unitex/src/fr/umlv/unitex/frames/GraphPathDialog.java [hunk 1]
```diff
@@ -64,6 +70,8 @@
 public class GraphPathDialog extends JDialog {
 	final BigTextList textArea = new BigTextList();
 	final JTextField graphName = new JTextField();
+	// issue #61
```
Comment (martinec) [inline unitex/src/fr/umlv/unitex/frames/GraphPathDialog.java]: It's not necessary to comment the issue here.

File: unitex/src/fr/umlv/unitex/frames/GraphPathDialog.java [hunk 2]
```diff
@@ -118,8 +126,10 @@ private JPanel constructPanel() {
 	}
 
 	private JPanel constructTopPanel() {
-		final JPanel top = new JPanel(new GridLayout(6, 1));
+		final JPanel top = new JPanel(new GridLayout(7, 1));
 		top.add(constructGraphNamePanel());
+		// issue #61
```
Comment (martinec) [inline unitex/src/fr/umlv/unitex/frames/GraphPathDialog.java]: It's not necessary to comment the issue here.

File: unitex/src/fr/umlv/unitex/frames/GraphPathDialog.java [hunk 3]
```diff
@@ -237,6 +266,51 @@ private JPanel constructGraphNamePanel() {
 		panel.add(graphName, BorderLayout.CENTER);
 		return panel;
 	}
+	
+	/**
+	 * Constructs panel with list file output location and browse button to allow user to select other file/location
+	 * issue #61
+	 * @return
```
Comment (martinec) [inline unitex/src/fr/umlv/unitex/frames/GraphPathDialog.java]: Omit `@return` for methods that return void and for constructors

File: unitex/src/fr/umlv/unitex/frames/GraphPathDialog.java [hunk 4]
```diff
@@ -237,6 +266,51 @@ private JPanel constructGraphNamePanel() {
...
+		
+		outputFileName.setText(ConfigManager.getManager()
+				.getCurrentLanguageDir() + String.valueOf(File.separatorChar) + "list.txt");
+		final JPanel panel = new JPanel(new BorderLayout());
+		panel.add(new JLabel(" Output file: "), BorderLayout.WEST);
+		panel.add(outputFileName, BorderLayout.CENTER);
+		
+		final JPanel button = new JPanel(new GridLayout(1, 1));
+		
+		final Action browseAction = new AbstractAction("Browse") {
```
Comment (martinec) [inline unitex/src/fr/umlv/unitex/frames/GraphPathDialog.java]: Use instead `Set File`

File: unitex/src/fr/umlv/unitex/frames/GraphPathDialog.java [hunk 5]
```diff
@@ -249,6 +323,20 @@ private JPanel constructGraphNamePanel() {
...
+				// issue #61 - recursive path option invokes UnitexToolLogger which hard codes the name of the output file to GraphNameautolst.txt 
+				// once that process has completed and loaded the file rename it using the user input if that differs from the default
+				if (!name.getAbsolutePath().equals(outputFileName.getText())) {
+					File dest = new File(outputFileName.getText());
+					Files.move(name.toPath(), dest.toPath(), StandardCopyOption.REPLACE_EXISTING);
+				}
+			}  catch (final IOException e) {
+				e.printStackTrace();
+				JOptionPane.showMessageDialog(null,
+						"Could not save file with name [" + outputFileName.getText() + "] a file has been created [" + name.getAbsolutePath() + "]", "Error",
```
Comment (martinec) [inline unitex/src/fr/umlv/unitex/frames/GraphPathDialog.java]: Given `/wrong/name.txt`, What about of  `Could not save path list to /wrong/name.txt`

File: unitex/src/fr/umlv/unitex/frames/GraphPathDialog.java [hunk 6]
```diff
@@ -180,10 +206,12 @@ public void actionPerformed(ActionEvent arg0) {
 				fst2 = new File(FileUtil.getFileNameWithoutExtension(graphName
```
Comment (rewopkram) [inline unitex/src/fr/umlv/unitex/frames/GraphPathDialog.java]: I added code to UnitexFrame.java to display error message if user clicks explore paths menu option before a graph has not been created

File: unitex/src/fr/umlv/unitex/frames/GraphPathDialog.java [hunk 7]
```diff
@@ -133,9 +143,25 @@ private JPanel constructTopPanel() {
...
+		// issue #61 add listeners to change default output file name based on user selection
+		onlyPaths.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent event) {
+				outputFileName.setText(ConfigManager.getManager()
+						.getCurrentLanguageDir() + String.valueOf(File.separatorChar) + "list.txt");
+			}
+		});
+		exploreRecursively.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent event) {
+				outputFileName.setText(FileUtil.getFileNameWithoutExtension(graphName
```
Comment (rewopkram) [inline unitex/src/fr/umlv/unitex/frames/GraphPathDialog.java]: Hi Cristian
Changed as requested

File: unitex/src/fr/umlv/unitex/frames/GraphPathDialog.java [hunk 8]
```diff
@@ -237,6 +266,51 @@ private JPanel constructGraphNamePanel() {
...
+	 * Constructs panel with list file output location and browse button to allow user to select other file/location
+	 * issue #61
+	 * @return
+	 */
+	private JPanel constructListFileNamePanel() {
+		
+		outputFileName.setText(ConfigManager.getManager()
+				.getCurrentLanguageDir() + String.valueOf(File.separatorChar) + "list.txt");
+		final JPanel panel = new JPanel(new BorderLayout());
+		panel.add(new JLabel(" Output file: "), BorderLayout.WEST);
```
Comment (rewopkram) [inline unitex/src/fr/umlv/unitex/frames/GraphPathDialog.java]: This is a hacky solution here. JLabel.setPreferredSize() didn't work as expected. I didn't want to change to FlowLayout without your approval.

File: unitex/src/fr/umlv/unitex/frames/GraphPathDialog.java (hunk 4 above)
Comment (rewopkram) [inline unitex/src/fr/umlv/unitex/frames/GraphPathDialog.java]: Changed as requested

File: unitex/src/fr/umlv/unitex/frames/GraphPathDialog.java [hunk 9]
```diff
@@ -237,6 +266,51 @@ private JPanel constructGraphNamePanel() {
...
+		};
+		final JButton browse = new JButton(browseAction);
+		button.add(browse);
+		panel.add(button, BorderLayout.EAST);
+		
+		return panel;
+	}
+	
+	private void openOutputFile() {
+		final int returnVal = Config.getTableDialogBox().showOpenDialog(this);
```
Comment (rewopkram) [inline unitex/src/fr/umlv/unitex/frames/GraphPathDialog.java]: Added change to Config.java

**Comment by rewopkram** [comment]
Do I need to issue another pull request for the updated changes?

**Comment by martinec** [comment]
LGTM

**Comment by martinec** [comment]
Thank you very much @rewopkram. You've got it made!

**Pull request merged**